pcsx2/ImGui/FullscreenUI.cpp

```diff
@@ -1756,21 +1763,60 @@ void FullscreenUI::DrawCustomBackground()
 			// Image is taller scale to width and crop top/bottom
 			scale = display_size.x / tex_width;
 		}
-		
+
 		const float scaled_width = tex_width * scale;
 		const float scaled_height = tex_height * scale;
 		const float offset_x = (display_size.x - scaled_width) * 0.5f;
 		const float offset_y = (display_size.y - scaled_height) * 0.5f;
-		
-		img_min = ImVec2(offset_x, offset_y);
-		img_max = ImVec2(offset_x + scaled_width, offset_y + scaled_height);
+
+		bg_draw_list->AddImage(tex_id,
+			ImVec2(offset_x, offset_y),
+			ImVec2(offset_x + scaled_width, offset_y + scaled_height),
+			ImVec2(0.0f, 0.0f), ImVec2(1.0f, 1.0f), col);
+	}
+	else if (mode == "center")
+	{
+		// Center image at original size
+		const float offset_x = (display_size.x - tex_width) * 0.5f;
+		const float offset_y = (display_size.y - tex_height) * 0.5f;
+
+		bg_draw_list->AddImage(tex_id,
+			ImVec2(offset_x, offset_y),
+			ImVec2(offset_x + tex_width, offset_y + tex_height),
+			ImVec2(0.0f, 0.0f), ImVec2(1.0f, 1.0f), col);
+	}
+	else if (mode == "tile")
```

Comment:
Found one last issue: If I use an extremely small image in tile mode, I get a crash because it tries to add a huge number of images to the draw list:
```
[  298.7816] Attempting to allocate 69939200 bytes from a 33554432 byte stream buffer
/home/thomas/projects/pcsx2/pcsx2/GS/Renderers/Vulkan/VKStreamBuffer.cpp:118: assertion failed in function bool VKStreamBuffer::ReserveMemory(u32, u32): Stream buffer overflow
```

Would it be possible to use a single image with carefully calculated UV coordinates instead? Off the top of my head I don't know if ImGui will use the right UV wrapping mode.

The image:
<img width="2" height="2" alt="echo" src="https://github.com/user-attachments/assets/3ef4c4fc-8bd3-4d81-8fc3-d5d0736fab70" />

Reply:
Just to check I know you said ImGui but this isn't for Qt right?

Reply:
will get it fixed then